src/UiPatternsManager.php

```diff
@@ -73,6 +73,7 @@ public function getDefinitions() {
         $definitions[$definition['id']] = $definition;
         unset($definitions[$id]);
       }
+      $definitions = $this->getSortedDefinitions($definitions);
```

Comment:
Doesn't this sort based on `category` value?

```php
  /**
   * {@inheritdoc}
   */
  public function getSortedDefinitions(array $definitions = NULL, $label_key = 'label') {
    // Sort the plugins first by category, then by label.
    /** @var \Drupal\Core\Plugin\CategorizingPluginManagerTrait|\Drupal\Component\Plugin\PluginManagerInterface $this */
    $definitions = isset($definitions) ? $definitions : $this->getDefinitions();
    uasort($definitions, function ($a, $b) use ($label_key) {
      if ($a['category'] != $b['category']) {
        return strnatcasecmp($a['category'], $b['category']);
      }
      return strnatcasecmp($a[$label_key], $b[$label_key]);
    });
    return $definitions;
  }
```

Since UI patterns do now have a category key they will all get the provider name:

```php
  protected function processDefinitionCategory(&$definition) {
    // Ensure that every plugin has a category.
    if (empty($definition['category'])) {
      // Default to the human readable module name if the provider is a module;
      // otherwise the provider machine name is used.
      $definition['category'] = $this->getProviderName($definition['provider']);
    }
  }
```

Will this have any effect on the UI?

Reply:
Yes, it sorts by category/provider first, but looking at the pattern definitions, the 'provider' key that it sorts by is always null since it accesses it directly off the class property instead of using the `getProvider()` method.

That being said, maybe this trait isn't quite the right fit since we're not making use of the categorization aspect of it. Maybe we just write our own sort method and call it a day?